eman2/protocols/protocol_tomo_extraction.py

```diff
@@ -112,30 +114,29 @@ def _defineParams(self, form):
     def _insertAllSteps(self):
         self._insertFunctionStep('writeSetOfCoordinates3D')
         self._insertFunctionStep('extractParticles')
+        self._insertFunctionStep('convertOutput')
         self._insertFunctionStep('createOutputStep')
 
     # --------------------------- STEPS functions -----------------------------
 
     def writeSetOfCoordinates3D(self):
         self.lines = []
         self.tomoFiles = []
-        inputSet = self.getInputTomograms()
-        for item in inputSet:
+        for tomo in self.getInputTomograms():
             coordDict = []
-            tomo = item.clone()
             self.coordsFileName = self._getExtraPath(
                 pwutils.replaceBaseExt(tomo.getFileName(), 'coords'))
 
             out = open(self.coordsFileName, "w")
-            for coord3DSet in self.inputCoordinates.get().iterCoordinates():
-                if os.path.basename(tomo.getFileName()) == os.path.basename(coord3DSet.getVolName()):
-                    out.write("%d\t%d\t%d\n" % (coord3DSet.getX(), coord3DSet.getY(), coord3DSet.getZ()))
-                    coordDict.append(coord3DSet.clone())
+            for coord3D in self.inputCoordinates.get().iterCoordinates(volume=tomo):
+                if os.path.basename(tomo.getFileName()) == os.path.basename(coord3D.getVolName()):
+                    out.write("%d\t%d\t%d\n" % (coord3D.getX(), coord3D.getY(), coord3D.getZ()))
+                    coordDict.append(coord3D.clone())
+            out.close()
```

Comment:
you could use "with open(file) as f" context manager, so that if you code fails the file will be closed.